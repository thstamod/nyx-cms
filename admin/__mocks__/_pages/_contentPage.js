export const sidebar = {
  data: {
    documentTypes: [
      {
        _id: '5e986ce27a141e6a5ca5f2ee',
        name: 'depth_0',
        __typename: 'DocumentType',
        descendants: [
          {
            documentType: {
              _id: '5e986cfb7a141e6a5ca5f2ef',
              name: 'depth_11',
              __typename: 'DocumentType',
              descendants: [],
            },
          },
          {
            documentType: {
              _id: '5e986d007a141e6a5ca5f2f0',
              name: 'depth12_with_child',
              __typename: 'DocumentType',
              descendants: [
                {
                  documentType: {
                    _id: '5e986d097a141e6a5ca5f2f1',
                    name: 'depth_2',
                    __typename: 'DocumentType',
                    descendants: [],
                  },
                },
              ],
            },
          },
        ],
      },
      {
        _id: '5e98743e707dfc01d6ce173e',
        name: 'depth0_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
      {
        _id: '5e989eee707dfc01d6ce173f',
        name: 'depth0_1_no child',
        __typename: 'DocumentType',
        descendants: [],
      },
    ],
  },
};
export const panel = {
  data: {
    documentType: {
      _id: '5e98743e707dfc01d6ce173e',
      name: 'depth0_no child',
      parentDocumentType: null,
      compilation: [
        {
          dataType: {
            type: 'Text',
            name: 'text',
          },
          value: {
            val: 'I am a Title!!',
          },
          title: 'title',
          options: null,
        },
        {
          dataType: {
            type: 'TextArea',
            name: 'TextArea',
          },
          value: {
            val:
              'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
          },
          title: 'big text',
          options: null,
        },
      ],
      __typename: 'DocumentType',
    },
  },
};
