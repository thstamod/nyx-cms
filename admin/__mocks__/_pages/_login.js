export default {
  data: {
    login: {
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZTcxNDczZjE4MTIxYzJmZjI0YzQwZWYiLCJlbWFpbCI6InRlc3RAbWFpbC5jb20iLCJpYXQiOjE1ODc4OTc3NTEsImV4cCI6MTU4NzkwMTM1MX0.3oIAITE622eRvD85ux1t5n1rLAhOJmyc7c7tTMXYgCA',
      tokenExpiration: 1,
    },
  },
};
