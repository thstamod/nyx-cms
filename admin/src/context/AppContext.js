import React, { createContext, useContext } from 'react';
import { userReducer, initialState } from '../redux/reducers/userReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const AppContext = createContext();

export function useAppState() {
  return useContext(AppContext);
}

export function AppStateProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(userReducer, initialState, middlewares);

  return <AppContext.Provider value={store}>{children}</AppContext.Provider>;
}
