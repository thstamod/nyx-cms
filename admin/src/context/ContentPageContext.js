import React, { createContext, useContext } from 'react';
import {
  initialState,
  contentPageReducer,
} from '../state/reducers/contentPageReducer';
import useEnhancedReducer from '../hooks/useEnhancedReducer';
import logger from '../middlewares/customLogger';

export const ContentPageContext = createContext(null);

export function useContentPageState() {
  return useContext(ContentPageContext);
}

export function ContentPageProvider({ children }) {
  const middlewares = [logger];
  const store = useEnhancedReducer(
    contentPageReducer,
    initialState,
    middlewares
  );

  return (
    <ContentPageContext.Provider value={store}>
      {children}
    </ContentPageContext.Provider>
  );
}
