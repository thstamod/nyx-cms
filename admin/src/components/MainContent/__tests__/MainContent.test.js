import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { ContentPageProvider } from '../../../context/ContentPageContext';
import MainContent from '../MainContent';
import { panel } from '../../../../__mocks__/_pages/_contentPage';

afterEach(cleanup);

test('MainContent renders', () => {
  const { asFragment } = render(
    <ContentPageProvider>
      <MockedProvider mocks={[]} addTypename={false}>
        <MainContent data={panel.data} />
      </MockedProvider>
    </ContentPageProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
