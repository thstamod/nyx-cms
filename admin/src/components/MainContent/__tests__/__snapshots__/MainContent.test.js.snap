// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders 1`] = `
<DocumentFragment>
  <div
    class="mainContent"
  >
    <h1>
      Content page
    </h1>
    <button
      type="button"
    >
      SAVE
    </button>
  </div>
</DocumentFragment>
`;
